ext/storage_sources/s3_store/aws_bucket_conn.cxx

```diff
@@ -100,6 +101,32 @@ aws_bucket_conn::delete_object(const std::string &bucket_name, const std::string
     }
 }
 
+/*
+ * object_exists --
+ *     Checks to see whether an object with the given key exists in the S3 bucket. Returns: 0 - The
+ *     object exists in the bucket. ENOENT - The object does not exist in the bucket. 1 - An error
+ *     has occurred.
+ */
+int
+aws_bucket_conn::object_exists(const std::string &bucket_name, const std::string &object_key) const
+{
+    Aws::S3Crt::Model::HeadObjectRequest request;
+    request.SetBucket(bucket_name);
+    request.SetKey(object_key);
+    Aws::S3Crt::Model::HeadObjectOutcome outcome = m_s3_crt_client.HeadObject(request);
+
+    if (outcome.IsSuccess())
+        return 0;
+    /*
+     * If an object with the given key does not exist the HEAD request will return a 404.
+     * https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html
+     */
+    else if (outcome.GetError().GetResponseCode() == Aws::Http::HttpResponseCode::NOT_FOUND)
+        return ENOENT;
+    else
+        return 1;
```

Comment:
What does 1 mean?

Reply:
Can we find an existing error code for that or just return the aws error code?

Reply:
The codes are defined here: https://github.com/aws/aws-sdk-cpp/blob/main/aws-cpp-sdk-core/include/aws/core/http/HttpResponse.h#L31

They are numeric, so I do not see a big issue returning them cast to an unsigned. We should also print an error message to stderr for now, later which will go into the wiredtiger logging stream. Eventually, it could be then an error code, along with a message. eg:

"401, Unauthorized."

Before we get there, this might be a good time to figure out how the functions will look like so that they all can follow the same pattern. For error handling, they all return an error code and a message.